### DISCUSSION TITLE
Limit q-table height to remaining vertical space

### DISCUSSION THREAD

**Discussion by kamcnally** [quasarframework/quasar#11396] [General - Components / Directives / etc] [answered]
What is the best way to limit a `q-table` height to fit within the available vertical space of a page so that the page is not scrollable but the `q-table` is? I want it to always fill the remaining space - so if there is no data then the `q-table` will be expanded to fit and if there is lots of data then the `q-table` should be scrollable.

I've created a CodePen [here](https://codepen.io/kamcnally/pen/qBXgrYx) of what I have managed to do.
The `q-header` has `q-toolbar` and `q-tabs` and the page has a row or two of various controls followed by the `q-table`.  For now, I've just set the table's parent `div` to have `height: 100vh; max-height: 100vh;`.

What I want            |  What I have
:-------------------------:|:-------------------------:
![Screenshot 2021-11-18 at 11 18 46](https://user-images.githubusercontent.com/879588/142405993-3ecb68cc-3277-4738-8de3-2e8067d866c6.png)  |  ![Screenshot 2021-11-18 at 11 06 18](https://user-images.githubusercontent.com/879588/142404159-d5cad1b2-b371-49e5-8808-4ae4369c0617.png)

**Comment by pdanpdan** [comment, marked as answer]
https://codepen.io/pdanpdan/pen/eYExyzJ?editors=1100

Ask if you have problems / do not understand what happens there / why some things are done.
